changelogs/unreleased/add-type-precedence-rules.yml

```diff
@@ -0,0 +1,7 @@
+---
+description: Add support to pass type precedence hints to the compiler
+issue-nr: 3098
+change-type: minor
+destination-branches: [master, iso5]
```

Comment:
Should this also go into ISO4?

Reply:
unfortunately too big for ISO4 . I'm afraid.